src/System.Runtime/tests/System/TimeZoneInfo.cs

```diff
@@ -10,6 +10,8 @@
 public static class TimeZoneInfoTests
 {
     private static readonly bool s_isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+    private static readonly bool s_isOSX = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
```

Comment:
I don't like breaking these into separate test methods because I think that breaks the cohesiveness of these checks.  In the case of NearMinMaxDateTimeOffsetConvertTest, all 3 branches in the if-else if-else statement are trying to test the same thing, only they all have different tz data to work with on the separate systems.

So instead, I broke the "data" out into a separate method that returns the correct offset for the timezone given the current platform.  Please have a look and tell me what you think.